shared/fs/browser/destination-picker/container.tsx

```diff
@@ -9,34 +9,35 @@ import * as FsGen from '../../../actions/fs-gen'
 import {isMobile} from '../../../constants/platform'
 import * as RouteTreeGen from '../../../actions/route-tree-gen'
 
-type OwnProps = RouteProps<
-  {
-    index: number
-  },
-  {}
->
+type OwnProps = {
+  index: number
+}
+type OwnPropsWithSafeNavigation = Container.PropsWithSafeNavigation<OwnProps>
 
 const mapStateToProps = (state: TypedState) => ({
   _destinationPicker: state.fs.destinationPicker,
   _pathItems: state.fs.pathItems,
 })
 
 type StateProps = ReturnType<typeof mapStateToProps>
-const getDestinationParentPath = memoize((stateProps: StateProps, ownProps: OwnProps) =>
-  stateProps._destinationPicker.destinationParentPath.get(
-    getRouteProps(ownProps, 'index') || 0,
-    stateProps._destinationPicker.source.type === Types.DestinationPickerSource.MoveOrCopy
-      ? Types.getPathParent(stateProps._destinationPicker.source.path)
-      : Types.stringToPath('/keybase')
-  )
+
+const getIndex = memoize((ownProps: OwnPropsWithSafeNavigation) => ownProps.getParam('index') || 0)
+const getDestinationParentPath = memoize(
```

Comment:
i don't think either of these should be memoized.
1. getIndex is fast so this'll just be slower and use more memory
2. getDestinationParentPath is memoizing on state and ownprops and both are likely mutating all the time